src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -42,6 +42,47 @@ private bool TryReadToSlow(out ReadOnlySpan<T> span, T delimiter, int skip, bool
             return true;
         }
 
+        /// <summary>
+        /// Try to read everything up to the given <paramref name="delimiter"/>, ignoring delimiters that are
+        /// preceeded by <paramref name="skipDelimiter"/>.
+        /// </summary>
+        /// <param name="span">The read data, if any.</param>
+        /// <param name="delimiter">The delimiter to look for.</param>
+        /// <param name="skipDelimiter">If found prior to <paramref name="delimiter"/> it will skip that occurence.</param>
+        /// <param name="advancePastDelimiter">True to move past the <paramref name="delimiter"/> if found.</param>
+        /// <returns>True if the <paramref name="delimiter"/> was found.</returns>
+        public bool TryReadTo(out ReadOnlySpan<T> span, T delimiter, T skipDelimiter, bool advancePastDelimiter = true)
```

Comment:
I think optionalDelimiterPrefix might be a better name

Reply:
For short variable names, delimiterEscape is fine.
If we are fine with a longer name: instead of prefixed with, how about `ignoreDelimiterIfEscapedWith`